webrtc/track.go

```diff
@@ -21,13 +21,32 @@ type RemoteTrack interface {
 	ReadRTP() (*rtp.Packet, error)
 }
 
+// RTPParameters represents RTCRtpParameters which contains information about
+// how the RTC data is to be encoded/decoded.
+//
+// ref: https://developer.mozilla.org/en-US/docs/Web/API/RTCRtpSendParameters
+type RTPParameters struct {
+	SSRC          uint32
+	SelectedCodec *webrtc.RTPCodec
+	Codecs        []*webrtc.RTPCodec
+}
+
 // RTPSender represents RTCRtpSender.
 type RTPSender interface {
 	ReplaceTrack(LocalTrack) error
+
+	// Parameters returns information about how the data is to be encoded.
+	Parameters() RTPParameters
```

Comment:
I like the current state. But, I feel a little bit awkward in using RTPParameters for giving codec availabilities, and choosing codec. For example, if I'm implementing the RTPSender's Parameters, the structure seems to tell me that I need to give "SSRC" and "SelectedCodec" values.

How about if we create a new type to differentiate between these 2 cases?

```go
type RTPSetParameters struct {
  SSRC uint32
  SelectedCodec *webrtc.RTPCodec
}

type RTPParameters struct {
  Codecs        []*webrtc.RTPCodec 
}
```

Reply:
I guess `PeerConnection` sets `RTPSetParameters` having multiple available codecs which the other peer supports, and RTPSender selects one of them? I have to read WebRTC specification bit more...

Reply:
According to https://w3c.github.io/webrtc-pc/#example-2 and https://w3c.github.io/webrtc-pc/#rtcrtpsender-interface (11), it seems to be
```go
type RTPCodec struct {
	webrtc.RTPCodec // v2 members
	Active bool
}
type RTPParameters struct {
	SSRC          uint32
	Codecs        []*webrtc.RTPCodec
}
```
and user may do something like
```go
p := sender.Parameters()
p.Codecs[automaticallySelectedCodecIndex].Active = false
p.Codecs[userSelectedCodecIndex].Active = true
sender.SetParameters(p)
```